pandas/io/excel/_util.py

```diff
@@ -23,32 +23,61 @@ def register_writer(klass):
     _writers[engine_name] = klass
 
 
-def get_default_writer(ext):
+def get_default_engine(ext, mode="reader"):
     """
-    Return the default writer for the given extension.
+    Return the default reader/writer for the given extension.
 
     Parameters
     ----------
     ext : str
         The excel file extension for which to get the default engine.
+    mode : str {'reader', 'writer'}
+        Whether to get the default engine for reading or writing.
+        Either 'reader' or 'writer'
 
     Returns
     -------
     str
         The default engine for the extension.
     """
+    _default_readers = {
+        "xlsx": "openpyxl",
+        "xlsm": "openpyxl",
+        "xlsb": "pyxlsb",
+        "xls": "xlrd",
+        "ods": "odf",
+    }
     _default_writers = {
         "xlsx": "openpyxl",
         "xlsm": "openpyxl",
+        "xlsb": "pyxlsb",
         "xls": "xlwt",
         "ods": "odf",
     }
-    xlsxwriter = import_optional_dependency(
-        "xlsxwriter", raise_on_missing=False, on_version="warn"
-    )
-    if xlsxwriter:
-        _default_writers["xlsx"] = "xlsxwriter"
-    return _default_writers[ext]
+    assert mode in ["reader", "writer"]
+    if mode == "writer":
+        # Prefer xlsxwriter over openpyxl if installed
+        xlsxwriter = import_optional_dependency(
+            "xlsxwriter", raise_on_missing=False, on_version="warn"
+        )
+        if xlsxwriter:
+            _default_writers["xlsx"] = "xlsxwriter"
+        return _default_writers[ext]
+    else:
+        if (
+            import_optional_dependency(
+                "openpyxl", raise_on_missing=False, on_version="warn"
+            )
+            is None
+            and import_optional_dependency(
+                "xlrd", raise_on_missing=False, on_version="raise"
```

Comment:
We shouldn't be raising here. If a user doesn't have openpyxl/xlrd installed, this will raise even if a user isn't reading an xlsx file. I don't think we should even be warning here (although I see that warnings are given for the writer code above); warnings/errors should only be raised if the outdated/not-installed package is attempted to be used.

Reply:
@jreback - thoughts?

Reply:
yeah this is solely to change the default.